### PR TITLE
[acceptance-tests] Add profiler-stress suite.

### PR DESCRIPTION
**Pull request by alexrp**
This currently consists of 44 benchmark programs from the [xamarin/benchmarker](https://github.com/xamarin/benchmarker) repository. The tests are executed with randomized profiler parameters in an effort to shake out bugs more effectively.

The tests take about 14 hours to execute on my machine (if all tests use the `alloc` option). However, since Jenkins bots will likely be a bit slower and the random nature of the tests can make them take longer, I've set the timeout to 20 hours.

### CODE REVIEW

**Comment by alexrp** [comment]
cc @akoeplinger

File: acceptance-tests/profiler-stress.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+SYS_REFS = \
+	System \
+	System.Core \
+	System.Data \
+	System.Runtime.Serialization \
+	System.Xml \
+	System.Xml.Linq
```
Comment (akoeplinger) [inline acceptance-tests/profiler-stress.mk]: These need to end with .dll

File: scripts/ci/run-test-profiler-stress-tests.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -e
+
+export TESTCMD=`dirname "${BASH_SOURCE[0]}"`/run-step.sh
+
+${TESTCMD} --label=check-profiler-stress --timeout=20h make -C acceptance-tests check-profiler-stress
```
Comment (akoeplinger) [inline scripts/ci/run-test-profiler-stress-tests.sh]: You need to add some logic to run-jenkins.sh so this file gets executed.

File: acceptance-tests/profiler-stress/runner.exe.sources [hunk 3]
```diff
@@ -0,0 +1,156 @@
+../../external/Newtonsoft.Json/Src/Newtonsoft.Json/Bson/BsonBinaryType.cs
```
Comment (akoeplinger) [inline acceptance-tests/profiler-stress/runner.exe.sources]: Not sure you need this, does -r:Newtonsoft.Json.dll not work?

File: acceptance-tests/profiler-stress.mk (hunk 1 above)
Comment (alexrp) [inline acceptance-tests/profiler-stress.mk]: Seems to be working fine without. What difference does this make?

File: acceptance-tests/profiler-stress/runner.exe.sources (hunk 3 above)
Comment (alexrp) [inline acceptance-tests/profiler-stress/runner.exe.sources]: `System.Net.Http.Formatting.dll` and various other assemblies in `mcs/` do it like this, and I can't find any makefiles that build and install `Newtonsoft.Json.dll` in the GAC.

Comment (akoeplinger) [inline acceptance-tests/profiler-stress/runner.exe.sources]: I thought we install it, but OK in that case :)

File: acceptance-tests/profiler-stress.mk (hunk 1 above)
Comment (akoeplinger) [inline acceptance-tests/profiler-stress.mk]: roslyn needs it, mcs doesn't, and we have been switching everything over to be consistent.

**Comment by alexrp** [comment]
@akoeplinger I think this addresses all your comments.

**Pull request merged**